tfhe/src/shortint/list_compression/compression.rs

```diff
@@ -57,6 +57,8 @@ impl CompressionKey {
                 let mut list: Vec<_> = vec![];
 
                 for ct in ct_list {
+                    assert!(ct.noise_level() == NoiseLevel::NOMINAL);
+
                     assert!(ct.carry_is_empty());
```

Comment:
could we have a nicer error message ? Also we may want to convert those asserts to errors at some point